http://gcc.gnu.org/ml/gcc-patches/2006-02/msg00177.html

2006-02-02  Jakub Jelinek  <jakub@redhat.com>

	* config/sparc/sparc.c (sparc_output_scratch_registers): Use
	#ignore instead of #scratch for %g7 .register directive.

Index: trunk/gcc/config/sparc/sparc.c
===================================================================
--- trunk/gcc/config/sparc/sparc.c	(revision 110521)
+++ trunk/gcc/config/sparc/sparc.c	(revision 110522)
@@ -3693,7 +3693,10 @@ sparc_output_scratch_registers (FILE *fi
 	  && ! sparc_hard_reg_printed [i])
 	{
 	  sparc_hard_reg_printed [i] = 1;
-	  fprintf (file, "\t.register\t%%g%d, #scratch\n", i);
+	  /* %g7 is used as TLS base register, use #ignore
+	     for it instead of #scratch.  */
+	  fprintf (file, "\t.register\t%%g%d, #%s\n", i,
+		   i == 7 ? "ignore" : "scratch");
 	}
       if (i == 3) i = 5;
     }
